lib/Controller/AttachmentController.php

```diff
@@ -81,6 +83,27 @@ public function __construct(
 		parent::__construct($appName, $request);
 	}
 
+	#[NoAdminRequired]
+	#[PublicPage]
+	#[RequireDocumentSessionUserOrShareToken]
+	public function getAttachmentList(?string $shareToken = null): DataResponse {
+		$documentId = $this->getDocument()->getId();
+		try {
+			$session = $this->getSession();
+		} catch (InvalidSessionException) {
+			$session = null;
+		}
+
+		if ($shareToken) {
+			$attachments = $this->attachmentService->getAttachmentList($documentId, null, $session, $shareToken);
```

Comment:
Not critical: Can we not skip the session (which should be null in this case anyways? would make it a bit clearer to read i think, maybe even using named arguments:

```suggestion
			$attachments = $this->attachmentService->getAttachmentList($documentId, shareToken: $shareToken);
```

Reply:
It's not said that session is null in this case. In public shares, when using `Editor.vue`, we still have a session. Only when using `MarkdownContentEditor.vue`, we don't have one.

Reply:
Right, missed that part. :)